osquery/tables/system/windows/intel_me.cpp

```diff
@@ -40,153 +47,892 @@ DEFINE_GUID(HECI_INTERFACE_GUID,
+  };
+  // clang-format on
+
+  auto it = kQueryFunctions.find(intel_me_info.interface_type);
+  if (it == kQueryFunctions.end()) {
+    return osquery::Status::failure("Invalid or unknown interface type");
+  }
+
+  const auto& query_function = it->second;
+  return query_function(intel_me_info, device);
+}
+
+osquery::Status queryIntelMEDeviceInterface(
+    IntelMEInformation& intel_me_info, const std::string& dev_interface_path) {
+  intel_me_info = {};
+
+  DeviceHandle device;
+  auto status = openDeviceInterface(device, dev_interface_path);
+  if (!status.ok()) {
+    return status;
+  }
+
+  status = queryHECIVersion(intel_me_info, device.get());
+  if (!status.ok()) {
+    return status;
+  }
+
+  status = connectToHECIInterface(intel_me_info, device.get());
+  if (!status.ok()) {
+    return status;
+  }
+
+  status = queryFirmwareVersion(intel_me_info, device.get());
+  if (!status.ok()) {
+    return status;
+  }
+
+  return osquery::Status::success();
+}
+} // namespace
+
+namespace tables {
+void getHECIDriverVersion(QueryData& results) {
+  results = {};
+
+  std::unordered_set<std::string> dev_path_list;
+  auto status = enumerateHECIDeviceInterfacePaths(dev_path_list);
+  if (!status.ok()) {
+    LOG(WARNING) << status.getMessage();
+    return;
+  }
+
+  if (dev_path_list.empty()) {
+    LOG(INFO) << "No Intel ME device found";
+    return;
+  }
+
+  for (const auto& dev_path : dev_path_list) {
+    IntelMEInformation intel_me_info;
+    status = queryIntelMEDeviceInterface(intel_me_info, dev_path);
+    if (!status.ok()) {
+      LOG(WARNING) << status.getMessage();
+      continue;
+    }
+
+    VLOG(1) << printIntelMEVersion(intel_me_info);
+
+    // Do not dump the whole structures; just get the essential version
+    // information
+    // and ignore the rest
+    std::string version = {};
+
+    switch (intel_me_info.interface_type) {
+    case IntelMEInformation::InterfaceType_0:
+    case IntelMEInformation::InterfaceType_1: {
+      const auto& fw_version =
+          boost::get<IntelMEInformation::FirmwareVersionTypes0And1>(
+              intel_me_info.fw_version);
+
+      version += std::to_string(fw_version.code.major) + "." +
+                 std::to_string(fw_version.code.minor) + "." +
+                 std::to_string(fw_version.code.hotfix) + "." +
+                 std::to_string(fw_version.code.build_number);
+
+      break;
+    }
+
+    case IntelMEInformation::InterfaceType_2: {
+      const auto& fw_version =
+          boost::get<IntelMEInformation::FirmwareVersionType2>(
+              intel_me_info.fw_version);
+
+      version += std::to_string(fw_version.major) + "." +
+                 std::to_string(fw_version.minor) + "." +
+                 std::to_string(fw_version.revision) + "." +
+                 std::to_string(fw_version.build);
+
+      break;
+    }
+
+    default: {
+      LOG(ERROR) << "Invalid interface type";
```

Comment:
This log line should provide more context, imagine if you are only looking at this single log line.